src/commands/commerce/extension/map.ts

```diff
@@ -60,7 +60,7 @@ export class MapExtension extends SfdxCommand {
         }
         const storeid = this.validateStoreId(storeName, storeId, userName);
         const registeredExternalServiceId = this.getRegisteredExtensionId(extensionName, userName);
-
+        this.deleteduplicateMaps(extensionName, userName);
```

Comment:
Is this function going to run even if there is no duplicate maps for an epn?
(Instead of having a separate function, why not have some checks in getRegisteredExtensionId and only have the deleting logic in deleteduplicateMaps maybe?)

Reply:
'getRegisteredExtensionId' runs only against the RegisteredExternalService table but the duplicates only exist in StoreIntegratedService table. So, it wouldn't be ideal to add extra queries and checks to include both tables in this method